docs/release-notes/august-2023.md

```diff
@@ -0,0 +1,157 @@
+:::
+This section of release notes includes upgrades of Semgrep OSS Engine for versions between **1.35.0** and **1.38.3**.
+
+### Added
+
+- The CLI now returns the commit timestamp when running `semgrep ci`.
+- Added optional `min-version` and `max-version` fields for a Semgrep rule, specifying a range of compatible Semgrep versions.
+    - If a rule is incompatible with the version of Semgrep being used, it is reported in the JSON output at the newly-added `info` level, which doesn't cause an exit failure. ([#8496](https://github.com/returntocorp/semgrep/pull/8496/))
+- The `semgrep scan` command is now more resilient to failures when fetching a configuration file (config) from Semgrep servers.
+    - If it can't fetch a config from Semgrep servers it will use backup infrastructure to fetch the most recent successful config for that customers environment. ([#8459](https://github.com/returntocorp/semgrep/pull/8459/))
+- `metavariable-comparison`: You can now use `in` and `not in` for strings in the same sense as in Python, for substring checking. ([#2979](https://github.com/returntocorp/semgrep/pull/8406))
+- **Julia:** Added the deep expression operator. Now you can write patterns such as `foo(<... 42 ...>)` to find instances of calls to `foo` that contain `42` somewhere inside of it. ([#8540](https://github.com/returntocorp/semgrep/pull/8540))
+- Added support for languages with case insensitive identifiers and generalized PHP to use these case insensitive identifiers.
+  - For example, in PHP the pattern `MyClass()` now matches calls with different capitalizations such as `myclass()` and `Myclass()`. ([#8356](https://github.com/returntocorp/semgrep/pull/8356))
+
+### Fixed
+
+- Fixed `--text` and `--output` flags which broke in 1.38.0. If you are using version 1.38.0, update Semgrep to receive these fixes.
+- Version 1.38.0 introduced some regressions with the Semgrep docker image. This is fixed in version 1.38.3.
+- Converted all '@r2c.dev' email addresses to '@semgrep.com'. Several error messages displayed outdated email addresses. With this fix, you can now see the correct email to reach out to the Semgrep Support team at [support@semgrep.com](mailto:support@semgrep.com). ([#8446](https://github.com/returntocorp/semgrep/pull/8446))
+- Fixed CLI output to display matches from different rules with the same message. Now you are able to see the rule ID granularly even if two rules have the same rule message. ([#8557](https://github.com/returntocorp/semgrep/pull/8557))
+- Semgrep PyPI package can now be installed on **aarch64 libmusl** platforms such as Alpine. (gh-8565)
+- Improved the `--max-memory` help description to make it clearer. Its previous message, "Defaults to 0 for all CLI scans" implies a different default for non-CLI scans, where in practicality the default is 0 for all scans except when using Pro Engine, where the default is 5000.
```

Comment:
This description is itself not quite right, I believe. @sinat101 Can you provide more context if I'm missing something? The default remains 0 for Pro Engine scans on CLI (just ran one myself on 1.38.3 to check). So presumably there are only certain environments where the 5000 default kicks in.

Reply:
@armchairlinguist I tried to check this for myself today - what command were you running to display the --max-memory? because I ran on verbose and nothing appeared 🤯

Reply:
Ah yes sorry, I should have said - it requires using --debug. I only ran it long enough for the semgrep-core invocation, since that's where you see which arguments get passed in.

Reply:
We got to the bottom of this case! Resolving.